### DISCUSSION TITLE
How get IP address from fields "addr_0", "addr_1"?

### DISCUSSION THREAD

**Discussion by ubushan** [cea-sec/ivre#1080] [Q&A] [answered]
Hello!
How to get IP address from fields "addr_0", "addr_1" collection "hosts" in MongoDB ? Or there are other way for get IP address?

![image](https://user-images.githubusercontent.com/12890432/106131369-a4dcd000-6173-11eb-90d5-7df50352dbf9.png)

Thanks.

**Comment by p-l-** [comment, marked as answer]
How do you get those results? MongoDB records are not supposed to be accessed outside of IVRE (they should only be used through the IVRE API, potentially exposed via the IVRE Web server).

The internal format used in the MongoDB storage (or other backends) may vary, and the Python functions keep the results consistent; this is what happened when the `"addr"` field has been changed to two "sub-fields" in the first place (so that we can have IPv6 addresses stored in IVRE).

If you use the standard `db.<purpose>.get(<filter>)` the IP address should be in the `"addr"` field, and the `"addr_0"` and `"addr_1"` fields should not exist (if that is not the case, it is a bug in IVRE).

**Reply by ubushan** [reply]
```
> db.views.find()[1]
        ...
	"addr_0" : NumberLong("-9223372036854775808"),
	"addr_1" : NumberLong("-9223090565858944068")
        ...
}


> db.hosts.find()[1]
        ...
	"addr_0" : NumberLong("-9223372036854775808"),
	"addr_1" : NumberLong("-9223090565858944068")
        ...
}

> db.views.find()[1]["addr"] # empty output
> db.hosts.find()[1]["addr"] # empty output
```

**Reply by p-l-** [reply]
You are using the **MongoDB shell**, and you should **not** do that. Rather, you should use the Python, backend-agnostic, API:
```Python
>>> from ivre.db import db
>>> cur = iter(db.view.get(db.view.flt_empty))
>>> next(cur)["addr"]
'1.2.3.4'
```
